### ISSUE TITLE
Ambiguous json type traits

### ISSUE THREAD

**Issue by xezon** [danielaparker/jsoncons#133]
json.hpp includes <jsoncons/json_type_traits.hpp>, so it forces implementations of type traits. When I try to implement my own type traits for a vector type, it will cause ambiguity with an existing type trait.

How is this intended to work?

**Comment by danielaparker** [comment]
Can you give an example of what you want to do? It's not clear to me why you would want to implement your own json type traits for the standard library containers.

**Comment by xezon** [comment]
I wanted to to have a container filled from an object, which is also somewhat array like. I can still do it manually, but can't use the type traits because it will see the existing type traits that works for json array -> container only.

**Comment by risksnap** [comment]
Okay, let me think about this.

**Comment by MavKillian** [comment]
How can I fix this? It is not compatible with latest Visual Studio? 

I am using Visual Studio Professional 2017, Version 15.7.5. I downloaded the JSONCONS zip file, unzipped and set my Config Properties "General Additional Include directories" to point to the jsoncons include directory. I then added all of the #include statements in my C++ program. However when I run Debug get many errors from the jsoncons "cbor_parser.hpp" include file such as:

Warning	C4244	'=': conversion from 'unsigned __int64' to 'size_t', possible loss of data	ConsoleApplication3	c:\users\killimm1\desktop\c code\jsoncons-master\jsoncons-master\include\jsoncons_ext\cbor\cbor_parser.hpp

Severity	Code	Description	Project	File	Line	Suppression State
Warning	C4244	'return': conversion from 'const __int64' to 'size_t', possible loss of data		

Severity	Code	Description	Project	File	Line	Suppression State
Error	C2664	'std::string jsoncons::cbor::detail::get_text_string(const uint8_t *,const uint8_t *,const uint8_t **)': cannot convert argument 3 from 'unsigned char **' to 'const uint8_t **'		

Severity	Code	Description	Project	File	Line	Suppression State
Warning	C4244	'argument': conversion from 'const unsigned __int64' to 'size_t', possible loss of data
